### PR TITLE
[ML] better handle empty results when evaluating regression

### PR DESCRIPTION
**Pull request by benwtrent**
If there are no documents that contain both the predicted and actual fields for the `regression` evaluation, we currently thrown an NPE because the aggregations contain `null` values. 

`regression` should match the `outlier` behavior and return an error stating that no docs exist.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/evaluation/regression/MeanSquaredError.java [hunk 1]
```diff
@@ -137,5 +137,17 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.endObject();
             return builder;
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            return error == error;
```
Comment (dimitris-athanasiou) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/evaluation/regression/MeanSquaredError.java]: Runny fingers! You got to cast and compare this and that :-)

Comment (benwtrent) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/evaluation/regression/MeanSquaredError.java]: They are always equal!!! :D

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
run elasticsearch-ci/1

**Pull request merged**